src/lib/services/types/tx.ts

```diff
@@ -17,72 +19,110 @@ import {
 // ----------------------------------------
 // --------------AuthInfo------------------
 // ----------------------------------------
-interface AuthInfo {
-  signer_infos: SignerInfo[];
-  fee?: Fee;
-}
+const zModeInfoMulti = z.object({
+  bitarray: z.object({
+    extraBitsStored: z.number(),
+    elems: z.array(z.number()),
+  }),
+  modeInfos: z.array(z.object({})),
+});
 
-interface SignerInfo {
-  public_key?: { "@type": string; key: string };
-  mode_info?: ModeInfo;
-  sequence: string;
-}
+const zModeInfoSingle = z.object({
+  mode: z.string(),
+});
 
-interface ModeInfo {
-  single?: ModeInfoSingle | undefined;
-  multi?: ModeInfoMulti | undefined;
-}
+const zModeInfo = z.object({
+  single: zModeInfoSingle.optional(),
+  multi: zModeInfoMulti.optional(),
+});
 
-interface ModeInfoSingle {
-  mode: keyof typeof SignMode;
-}
+const zSignerInfo = z
+  .object({
+    public_key: z.object({
+      "@type": z.string(),
+      key: z.string(),
+    }),
+    mode_info: zModeInfo.optional(),
+    sequence: z.string(),
+  })
+  .transform(snakeToCamel);
 
-// Revisit Multisig Info
-interface ModeInfoMulti {
-  bitarray?: CompactBitArray;
-  modeInfos: ModeInfo[];
-}
+const zAuthInfo = z
+  .object({
+    signer_infos: z.array(zSignerInfo),
+    fee: z.object({
+      amount: z.array(zCoin),
+      gas_limit: z.string(),
+      payer: z.string(),
+      granter: z.string(),
+    }),
+  })
+  .transform(snakeToCamel);
+export type AuthInfo = z.infer<typeof zAuthInfo>;
 
 // ----------------------------------------
 // -----------------Tx---------------------
 // ----------------------------------------
-export interface MsgBody {
-  "@type": TypeUrl;
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  [key: string]: any;
-}
+const zTxBody = z
+  .object({
+    messages: z.array(zMessageResponse).nullable(),
+    memo: z.string(),
+    timeout_height: z.string(),
+    extension_options: z.any().array(),
```

Comment:
Create zAny for this that convert to `import type { Any } from "cosmjs-types/google/protobuf/any";`